File: LeetCode/Math/JS/Leetcode883-三维形体投影面积.js
var projectionArea = function(grid) {
    const n = grid.length;
    let [l,r,d] = [0,0,0];
    for(let i = 0;i < n;i++) {
        let [tl,tr] = [0,0];
        for(let j = 0;j < n;j++) {
            if(grid[i][j]) d++;
            tl = Math.max(tl,grid[i][j]);
            tr = Math.max(tr,grid[j][i]);
        }
        l += tl,r += tr;
    }
    return l + r + d;
};

File: LeetCode/Math/Golang/Leetcode883-三维形体投影面积.go
package sbgo

func projectionArea(grid [][]int) int {
	n, l, r, d := len(grid), 0, 0, 0
	for i := 0; i < n; i++ {
		tl, tr := 0, 0
		for j := 0; j < n; j++ {
			if grid[i][j] != 0 {
				d++
			}
			tl = max(tl, grid[i][j])
			tr = max(tr, grid[j][i])
		}
		l += tl
		r += tr
	}
	return l + r + d
}

func max(i, j int) int {
	if i < j {
		return j
	}
	return i
}
